packages/appservice-rest/src/Utilities/KuduServiceUtility.ts

```diff
@@ -37,15 +37,16 @@ export class KuduServiceUtility {
         return deploymentID;
     }
 
-    public async deployUsingZipDeploy(packagePath: string): Promise<string> {
+    public async deployUsingZipDeploy(packagePath: string, customMessage?: any): Promise<string> {
         try {
             console.log('Package deployment using ZIP Deploy initiated.');
 
             let queryParameters: Array<string> = [
                 'isAsync=true',
                 'deployer=' + GITHUB_ZIP_DEPLOY
             ];
-
+            var deploymentMessage = this._getUpdateHistoryRequest(null, null, customMessage).message;
+            queryParameters.push('message=' + encodeURIComponent(deploymentMessage));

```

Comment:
should we push this param only when deploymentMessage is not empty

Reply:
deploymentMessage will never be empty please check this return statement

https://github.com/microsoft/pipelines-appservice-lib/blob/272b57f3993cba1edf87fff6bfcce730fdf167f9/packages/appservice-rest/src/Utilities/KuduServiceUtility.ts#L191